aws-datastore/src/androidTest/java/com/amplifyframework/datastore/appsync/SynchronousAppSync.java

```diff
@@ -107,11 +123,31 @@ public static SynchronousAppSync using(@NonNull AppSync appSync) {
      * @return Response data from AppSync.
      * @throws DataStoreException On failure to obtain response data
      */
+    @SuppressWarnings("LineLength")
     @NonNull
     <T extends Model> GraphQLResponse<ModelWithMetadata<T>> delete(
-            @NonNull Class<T> clazz, @NonNull String objectId, @NonNull Integer version) throws DataStoreException {
+        @NonNull Class<T> clazz, @NonNull String objectId, @NonNull Integer version) throws DataStoreException {
+        return delete(clazz, objectId, version, null);
+    }
+
+    /**
+     * Uses Amplify API to make a mutation which will only apply if the version sent matches the server version.
+     * @param clazz The class of the object being deleted
+     * @param objectId ID id of the object to delete
+     * @param version The version of the model we have
+     * @param predicate The condition to be applied to the delete.
+     * @param <T> The type of data in the response. Must extend Model.
+     * @return Response data from AppSync.
+     * @throws DataStoreException On failure to obtain response data
+     */
+    @SuppressWarnings("LineLength")
```

Comment:
Anyway to avoid this? (newline before the `throws`?)

Reply:
We can swing the params down onto a new line.

My personal style is to indent +8 off the method block, which differentiates it from the method body, and the method signature.

```
@NonNull
<T extends Model> GraphQLResponse<ModelWithMetadata<T>> delete(
        @NonNull Class<T> clazz,
        @NonNull String objectId,
        @NonNull Integer version,
        @Nullable QueryPredicate predicate)
        throws DataStoreException {
    doStuffHereAtPlus4();
    coolStuffOnly();
    thatIs();
}
```

The rule is basically "Chomp down vars in a method signature, and indent by 8."

I know of _no tool_ that is able to uniquely distinguish this case, from other line-wrapping or indentation cases.

My personal style is to use +4 spaces as indentation in all other situations in a Java document.

Well,
```
final class MyClass implements
        SameTypeOfThing,
        Here {
    public void soThatYouCanTellItIsntThisMethodEasily() {
    }
}
```